host/resource/resource.go

```diff
@@ -0,0 +1,54 @@
+package resource
+
+import . "github.com/flynn/flynn/Godeps/_workspace/src/github.com/docker/docker/pkg/units"
+
+type Spec struct {
+	// Request is the amount of resource a job expects to consume, so the
+	// job should only be placed on a host with at least this amount of
+	// resource available, and once scheduled this amount of resource
+	// should then be unavailable on the given host.
+	Request int64 `json:"request"`
+
+	// Limit is an upper limit on the amount of resource a job can consume,
+	// the outcome of hitting this limit being implementation defined (e.g.
+	// a system error, throttling, catchable / uncatchable signals etc.)
+	Limit int64 `json:"limit"`
+}
+
+type Type string
+
+const (
+	TypeMemory   Type = "memory"
+	TypeMaxFD    Type = "max_fd"
+	TypeMaxProcs Type = "max_procs"
+)
+
+var defaults = Resources{
+	TypeMemory:   {Request: 1 * GiB, Limit: 1 * GiB},
+	TypeMaxFD:    {Request: 10000, Limit: 10000},
+	TypeMaxProcs: {Request: 256, Limit: 256},
+}
+
+type Resources map[Type]Spec
+
+func Defaults() Resources {
+	r := make(Resources)
+	SetDefaults(&r)
+	return r
+}
+
+func SetDefaults(r *Resources) {
+	if *r == nil {
+		*r = make(Resources)
+	}
+	for typ, s := range defaults {
+		spec := (*r)[typ]
+		if spec.Limit == 0 {
+			spec.Limit = s.Limit
+		}
+		if spec.Request == 0 {
```

Comment:
Zero seems like a valid value for some resource types. Does it need to be a pointer?